app/controllers/users/dossiers_controller.rb

```diff
@@ -558,6 +558,10 @@ def update_dossier_and_compute_errors
       @dossier.update_champs_attributes(champs_public_attributes_params, :public, updated_by: current_user.email)
       if @dossier.champs.any?(&:changed_for_autosave?)
         @dossier.last_champ_updated_at = Time.zone.now
+
+        if @dossier.champs_public.filter(&:changed_for_autosave?).any?(&:used_by_routing_rules?)
```

Comment:
Idem, me semble plus lisible, mais a ne pas te freiner pr merger si tu as d'autre gout
```suggestion
        if @dossier.champs_public.any? { _1.changed_for_autosave? && _1.used_by_routing_rules? }
```